packages/gatsby/src/utils/webpack-status.ts

```diff
@@ -10,9 +10,10 @@ export function markWebpackStatusAsPending(): void {
   isPendingStatus = true
 }
 
-export function markWebpackStatusAsDone(): void {
+export async function markWebpackStatusAsDone(): Promise<void> {
   isPendingStatus = false
+
   if (isFlushEnqueued()) {
-    flush()
+    await flush()
```

Comment:
Adding await wasn't necessary to fix the bug but this seems like a possible timing issue that can occur. Maybe we don't care about awaiting here but than I would like to add a different function that flushes if necessary.

During `gatsby build` we never wait until this possible flush is done. I can't think of when we ever have to flush but this seems like a code smell to me.